File: modlib/src/main/java/heroes/journey/modlib/Position.java
package heroes.journey.modlib;

public class Position {

    private int x, y;

    public Position(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public Position setX(int x) {
        this.x = x;
        return this;
    }

    public Position setY(int y) {
        this.y = y;
        return this;
    }

    public Position setPos(int x, int y) {
        this.x = x;
        this.y = y;
        return this;
    }

    public String toString() {
        return "(" + x + ", " + y + ")";
    }

    public int distanceTo(Position other) {
        int dx = this.x - other.x;
        int dy = this.y - other.y;
        // return (int) Math.sqrt(dx * dx + dy * dy); // Euclidean
        return Math.abs(this.x - other.x) + Math.abs(this.y - other.y); // Manhattan
    }

    public int getX() {
        return this.x;
    }

    public int getY() {
        return this.y;
    }

    public void reset() {
        x = -1;
        y = -1;
    }
}

File: modlib/src/main/java/heroes/journey/modlib/utils/Position.kt
package heroes.journey.modlib.utils

import kotlin.math.abs

class Position(@JvmField var x: Int, @JvmField var y: Int) {
    fun setX(x: Int): Position {
        this.x = x
        return this
    }

    fun setY(y: Int): Position {
        this.y = y
        return this
    }

    fun setPos(x: Int, y: Int): Position {
        this.x = x
        this.y = y
        return this
    }

    override fun toString(): String {
        return "($x, $y)"
    }

    fun distanceTo(other: Position): Int {
        val dx = this.x - other.x
        val dy = this.y - other.y
        // return (int) Math.sqrt(dx * dx + dy * dy); // Euclidean
        return (abs((this.x - other.x).toDouble()) + abs((this.y - other.y).toDouble())).toInt() // Manhattan
    }

    fun reset() {
        x = -1
        y = -1
    }
}
